setup.py

```diff
@@ -162,7 +162,7 @@
     "sacremoses",
     "safetensors>=0.4.1",
     "sagemaker>=2.31.0",
-    "schedulefree>=1.2.6",
+    "schedulefree>=1.4",
```

Comment:
can we not update the minimum version of schedulefree here but add checks in trainer.py ?